ext/http/request_body.rs

```diff
@@ -36,6 +36,9 @@ impl Stream for ReadFuture {
           }
         }
       }
+      std::task::Poll::Ready(Some(Err(e))) => {
+        return std::task::Poll::Ready(Some(Err(e.into())));
+      }
       std::task::Poll::Ready(None) => return std::task::Poll::Ready(None),
       _ => {}
```

Comment:
Bug caused by catch-all?

Reply:
I agree, let's add all arms explicitly

Reply:
Good call. I will make it more explicit.

Reply:
Rewrote this function as it had a chance to lose the waker as well